### ISSUE TITLE
Cython.Coverage produces error when using gevent

### ISSUE THREAD

**Issue by wjsi** [cython/cython#2776]
Hi,

I build a simple pyx with ``gevent`` module (https://github.com/wjsi/test_cython_cov/blob/master/test_cov.pyx) and try to use Cython.Coverage to get code coverage, it produces an error when ``coverage report`` is executed:

```
$ coverage run test_file.py
$ coverage report
Plugin 'Cython.Coverage.Plugin' did not provide a file reporter for '/Users/wenjun.swj/miniconda3/lib/python3.7/site-packages/gevent/_hub_local.py'.
```

When gevent is removed, for instance, replaced with ``time`` module, no errors raised and the report is shown normally.

I uploaded all test files in https://github.com/wjsi/test_cython_cov. I'm running on MacOS 10.14 with Anaconda Python 3.7.1, cython==0.29.2 and coverage==4.5.2.

**Comment by scoder** [comment]
Thanks for the report. I have no idea what gevent does here, so this needs more investigation to find out how this differs from normal CPython behaviour and what can be done to make this work.

**Comment by wjsi** [comment]
@scoder during compilation, gevent compiles its _hub_local.py in https://github.com/gevent/gevent/blob/master/setup.py#L117 using cython into __hub_local.so and then import the compiled lib on the bottom of https://github.com/gevent/gevent/blob/master/src/gevent/_hub_local.py#L101 to get acceleration, which produces a _hub_local.c with path gevent/src/gevent/_hub_local.c. However, when installed as a wheel package, the path becomes gevent/_hub_local.c, therefore Cython.Coverage cannot spot the path of _hub_local.py. When creating the tracer, it reports no errors. However, when creating the reporter, no source file can be spotted and the error occurs.
I've written a test case to reproduce this error in #2784.

**Comment by user36362992** [comment]
I'm also having this issue:

`WARNING: Failed to generate report: Plugin 'Cython.Coverage.Plugin' did not provide a file reporter for u'/opt/opsview/python/lib/python2.7/site-packages/gevent/greenlet.py'.`

Any news on when/how it might be resolved?

**Comment by scoder** [comment]
I think this qualifies for a backport.

**Comment by scoder** [comment]
Actually, the fix was included in 0.29.3, so this is resolved already.
